### ISSUE TITLE
question about the gradient.

### ISSUE THREAD

**Issue by qiguming** [YuxinWenRick/diffusion_memorization#7]
I have question about the following code. I think the  prompt_embeds[[0], :, :] is **negative_prompt embedding** and prompt_embeds[[-1], :, :] is **positive_prompt embedding**. However, why the positive_prompt is placed first in the final input_prompt_embeds.
  
single_prompt_embeds = prompt_embeds[[0], :, :].clone().detach()
single_prompt_embeds.requires_grad = True
dummy_prompt_embeds = prompt_embeds[[-1], :, :].clone()
input_prompt_embeds = torch.cat(
      [
           dummy_prompt_embeds.repeat(num_images_per_prompt, 1, 1),
            single_prompt_embeds.repeat(num_images_per_prompt, 1, 1),
       ]
 )

**Comment by YuxinWenRick** [comment]
Hi, thanks for your question, and sorry about the late response. I think you are right, the order is reversed. However, I don't think this matters for the norm visualization, since we use l2 norm for the metric, and it's symmetric, so it should be fine.

Let me know if you have further questions!
